ci/datamgr.py

```diff
@@ -645,38 +617,84 @@ def impala(data_dir):
             port=env.hdfs_port,
             user=env.hdfs_user,
         ),
-        pool_size=CPU_COUNT,
+        pool_size=URLLIB_DEFAULT_POOL_SIZE,
     )
 
-    # validate our environment before performing possibly expensive operations
-    impala_raise_if_cannot_write_to_hdfs(con, env)
+    fs = fsspec.filesystem("file")
+    data_files = fs.find(data_dir)
 
-    # load the data files
-    load_impala_data(con, str(data_dir), env)
+    executor = ctx.obj["executor"]
 
-    # build and upload the UDFs
-    impala_build_udfs()
-    impala_upload_udfs(con, env)
+    hdfs = con.hdfs
+    # make the database and build and upload UDFs and data files
+    for future in concurrent.futures.as_completed(
+        {
+            executor.submit(impala_create_test_database, con, env),
+            *impala_build_and_upload_udfs(hdfs, env, fs=fs, executor=executor),
```

Comment:
Could you return the `(callback, args)` from `impala_build_and_upload_udfs` and call `executor.submit()` on those tasks instead?